modules/lang-painless/src/main/java/org/elasticsearch/painless/CompilerSettings.java

```diff
@@ -19,10 +19,18 @@
 
 package org.elasticsearch.painless;
 
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.Property;
+
 /**
  * Settings to use when compiling a script.
  */
 public final class CompilerSettings {
+    /**
+     * Are regexes enabled? This is a node level settings because regexes break out of painless's lovely sandbox and can cause stack
```

Comment:
Just to note, we do catch StackOverflows and consider that okay to happen within the sandbox.  It won't bring down a node or anything like that.